src/Tools/dotnet-monitor/CollectionRules/Actions/GetEnvironmentVariableAction.cs

```diff
@@ -0,0 +1,90 @@
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions
+{
+    internal sealed partial class GetEnvironmentVariableActionFactory :
+        ICollectionRuleActionFactory<GetEnvironmentVariableOptions>
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ILogger<GetEnvironmentVariableActionFactory> _logger;
+
+        public GetEnvironmentVariableActionFactory(IServiceProvider serviceProvider, ILogger<GetEnvironmentVariableActionFactory> logger)
+        {
+            _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            _logger = logger ?? throw new ArgumentNullException(nameof(serviceProvider));
+        }
+
+        public ICollectionRuleAction Create(IEndpointInfo endpointInfo, GetEnvironmentVariableOptions options)
+        {
+            if (null == options)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            ValidationContext context = new(options, _serviceProvider, items: null);
+            Validator.ValidateObject(options, context, validateAllProperties: true);
+
+            return new GetEnvironmentVariableAction(_logger, endpointInfo, options);
+        }
+
+        internal sealed partial class GetEnvironmentVariableAction :
+            CollectionRuleActionBase<GetEnvironmentVariableOptions>
+        {
+            private readonly IEndpointInfo _endpointInfo;
+            private readonly ILogger _logger;
+            private readonly GetEnvironmentVariableOptions _options;
+
+            public GetEnvironmentVariableAction(ILogger logger, IEndpointInfo endpointInfo, GetEnvironmentVariableOptions options)
+                : base(endpointInfo, options)
+            {
+                _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+                _endpointInfo = endpointInfo ?? throw new ArgumentNullException(nameof(endpointInfo));
+                _options = options ?? throw new ArgumentNullException(nameof(options));
+            }
+
+            protected override async Task<CollectionRuleActionResult> ExecuteCoreAsync(
+                TaskCompletionSource<object> startCompletionSource,
+                CancellationToken token)
+            {
+                try
+                {
+                    DiagnosticsClient client = new DiagnosticsClient(_endpointInfo.Endpoint);
+
+                    _logger.GettingEnvironmentVariable(_options.Name, _endpointInfo.ProcessId);
+                    Dictionary<string, string> envBlock = await client.GetProcessEnvironmentAsync(token);
+                    envBlock.TryGetValue(Options.Name, out string value);
```

Comment:
If this fails, it probably should throw an exception with some human readable text e.g. `The environment block does not contain the '{0}' variable.`. In the future, we could add an optional setting that allows silent failure and to report an empty value.